test/DurableTaskEndToEndTests.cs

```diff
@@ -107,11 +130,48 @@ public DurableTaskEndToEndTests(ITestOutputHelper output)
                 await host.StartAsync();
 
                 var client = await host.StartOrchestratorAsync(orchestratorFunctionNames[0], "World", this.output);
-                var status = await client.WaitForCompletionAsync(TimeSpan.FromSeconds(30), this.output);
+                var status = await client.WaitForCompletionAsync(TimeSpan.FromSeconds(30), this.output, showHistory, showHistoryOutput);
 
                 Assert.Equal(OrchestrationRuntimeStatus.Completed, status?.RuntimeStatus);
                 Assert.Equal("World", status?.Input);
                 Assert.Equal("Hello, World!", status?.Output);
+                if (!showHistory)
+                {
+                    Assert.Equal(null, status?.History);
+                }
+                else
+                {
+                    Assert.Equal(3, status?.History.Count);
+                    Assert.Equal<string>("ExecutionStarted", status?.History[0]["EventType"].ToString());

```

Comment:
Rather than using `status?` everywhere, we should just `Assert.NotNull(status)` earlier in this method.